ckan/lib/search/query.py

```diff
@@ -366,6 +366,8 @@ def run(self, query, permission_labels=None, **kwargs):
             query['mm'] = query.get('mm', '2<-1 5<80%')
             query['qf'] = query.get('qf', QUERY_FIELDS)
 
+        query.setdefault("df", "text")
```

Comment:
I think that if it's possible to do this in solr config, that would be ideal, as it reduces the impact on ckan core.

Reply:
Have no objections, but it's up to Adria :)

In order to remove these lines, we need to update `solrconfig.xml` inside the core's `conf` folder and set the following `defaults` for the request handler
```
<str name="df">text</str>
<str name="q.op">AND</str>
```

Reply:
The key though is that then CKAN can work, unmodified, with either solr backend.  This decouples the CKAN upgrade path from the solr upgrade path, at least for the current set of upgrades.  It also allows backports on the CKAN side without requiring that you update existing infra in a point release.

Reply:
I'm not opposed to shifting these changes to the Solr config side (although I have a to say that after going through a manual install of Solr it is not the most friendly process for a beginner user). I think having a clearer picture of the steps involved in installing and setting up Solr and write the install instructions for users will help me decide what we should do, so bear with me.

I guess that if we want to support both versions of Solr now, the alternative is to query the Solr version at startup time (like we do to get the schema version) using [this endpoint](http://localhost:8983/solr/admin/info/system) and conditionally execute these lines but is pretty ugly tbh and probably adds too much complexity for saving users to edit one file.

@EricSoroos can you clarify what you mean with this?
>  It also allows backports on the CKAN side without requiring that you update existing infra in a point release.

If you want to run CKAN 2.9 against Solr 8 you will need to do the changes outlined in this PR, so you'll need to update your existing infra right?

Reply:
It Allows. It doesn't Require.

* I think that requiring a solr8 upgrade in a point/bugfix release is a non-starter.
* I think that providing the option to upgrade solr outside of a minor/major release cycle is a good idea. 

So. IFF we can manage CKAN running against solr6 or solr8 with no changes on the CKAN side, then it's something that we should prioritize because it simplifies the backport process. Bring up a new solr server, configure it, point ckan at it, and rebuild indexes. If we can't, we can't. (but I don't think that's the case)

FWIW, I'm running solr8.8 in two instances, both of them are dev/testing, but they're working with unmodified ckan 2.9. I'm not going to claim 100% compatibility due to a lack of exhaustive testing, but I haven't noticed anything of an issue other than setting things up in the first place. 

For reference, my standard solr install in 6 is essentially: Pull solr6:, slap in some default config files in the docker image, then add a volume for data. Potentially bind mount in a new schema.xml on particular instances, depending.

For solr 8, because of the upstream docker image layout changes, you can't actually put a set of core config files in the image, you have to add them as a volume later. So I have something like an initdb_solr stage in the makefile, similar to the initdb stages that are required for postgres.  Inlining a config file in addition to the schema here is just one more file to have in the volume that gets added.